packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -827,6 +828,16 @@ export class CloudFrontWebDistribution extends cdk.Construct implements IDistrib
             lambdaFunctionArn: fna.lambdaFunction && fna.lambdaFunction.functionArn,
           }))
       });
+
+      // allow edgelambda.amazonaws.com to assume the functions' execution role.
+      for (const a of input.lambdaFunctionAssociations) {
+        if (a.lambdaFunction.role && a.lambdaFunction.role instanceof iam.Role && a.lambdaFunction.role.assumeRolePolicy) {
```

Comment:
Ugh that use of `instanceof` 🤭. Don't we have a `Rome.isRole`?

Reply:
Also - I don't see the test for the case where the function doesn't have a role or has one you can't add statements to... 😇

Reply:
`instanceof` is ok because we require a single version line through peer deps, so the issue `isXxx` addressed is no longer a problem.

I'll add a bunch of unit tests.